lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3394,6 +3394,87 @@ class ConvertAtenNumelOp : public OpConversionPattern<AtenNumelOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenIndexSelectOp : public OpConversionPattern<AtenIndexSelectOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenIndexSelectOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op.getLoc();
+    Value input = adaptor.self();
+    Value indices = adaptor.index();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType indicesType = indices.getType().cast<RankedTensorType>();
+    RankedTensorType resultType = getTypeConverter()
+                                      ->convertType(op->getResult(0).getType())
+                                      .cast<RankedTensorType>();
+    Type elementType = resultType.getElementType();
+    unsigned inputRank = inputType.getRank();
+
+    int64_t dimInt;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dimInt)))
+      return op->emitError("unimplemented: dim is not constant");
+
+    Value initTensor;
+    if (inputType.hasStaticShape() && indicesType.hasStaticShape()) {
```

Comment:
why is there a special case for static shape? We generally avoid that (except possibly for static size-1 dimensions) in TorchToLinalg.